kpi/tests/api/v2/test_api_assets.py

```diff
@@ -439,6 +439,61 @@ def test_report_styles_field(self):
             test_data=test_data
         )
 
+    def test_report_submissions(self):
+        # Prepare the mock data
+        report_url = reverse(
+            self._get_endpoint('asset-reports'), kwargs={'uid': self.asset_uid}
+        )
+        anotheruser = User.objects.get(username='anotheruser')
+        self.asset.content = {"survey": [
+            {"type": "select_one", "label": "q1", "select_from_list_name": "iu0sl99",},
+        ]
+        , "choices": [
+            {"name": "a1", "label": ["a1"], "list_name": "iu0sl99"},
+            {"name": "a3", "label": ["a3"], "list_name": "iu0sl99"},
+        ]}
+        self.asset.save()
+        self.asset.deploy(backend='mock', active=True)
+        submissions = [
+            {
+                "__version__": self.asset.latest_deployed_version.uid,
+                "q1": "a1",
+                "_submitted_by": "",
+            },
+            {
+                "__version__": self.asset.latest_deployed_version.uid,
+                "q1": "a3",
+                "_submitted_by": "",
+            },
+        ]
+
+        self.asset.deployment.mock_submissions(submissions)
+        partial_perms = {
+            PERM_VIEW_SUBMISSIONS: [{'_submitted_by': anotheruser.username}]
+        }
+        self.asset.assign_perm(
+            anotheruser, PERM_PARTIAL_SUBMISSIONS, partial_perms=partial_perms
+        )
+        # Verify endpoint works with the asset owner
+        response = self.client.get(report_url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        # Verify the endoiubt returns data
+        self.assertEqual(response.data['count'], 1)
+        # Verify the endpoint request fails when the user is logged out
+        self.client.logout()
+        response = self.client.get(report_url)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        # Verify a user with partial view permissions can access the data
+        self.client.login(username='anotheruser', password='anotheruser')
```

Comment:
You can use the same user with different permissions. 
We should test with `'view_submissions'` and `'partial_submissions'` 

Reply:
Fixed!